src/pages/Swap.tsx

```diff
@@ -277,9 +282,9 @@ function Swap(): ReactElement {
         formStateArg.swapType === SWAP_TYPES.TOKEN_TO_SYNTH &&
         bridgeContract != null
       ) {
-        const originPool = POOLS_MAP[formStateArg.from.poolName]
+        const originPool = basicPools?.[formStateArg.from.poolName]
         amountToReceive = await bridgeContract.calcTokenToSynth(
-          originPool.addresses[chainId],
+          originPool?.metaSwapDepositAddress ?? "",
```

Comment:
Defaulting to "" or 0 doesn't make sense for any of the values here and below - `calcTokenToSynth` will fail without a valid address, and all math will be wrong if we default a token's decimals to anything. If the pool or token is undefined, it's not even worth trying to execute the function that depends on those things.